File: content/python/templates/factorial.py
# Factorial Calculation

def factorial(n):
    """Calculate factorial of n."""
    if n < 0:
        return None
    if n <= 1:
        return 1
    return n * factorial(n - 1)

# Iterative version
def factorial_iter(n):
    """Calculate factorial iteratively."""
    result = 1
    for i in range(2, n + 1):
        result *= i
    return result

# Test
n = 5
print(f"{n}! = {factorial(n)}")
print(f"{n}! = {factorial_iter(n)}")

File: content/java/templates/06_Algorithms/factorial.java
// Factorial Calculation

public class Factorial {
    // Recursive version
    public static long factorial(int n) {
        if (n < 0) return -1;
        if (n <= 1) return 1;
        return n * factorial(n - 1);
    }
    
    // Iterative version
    public static long factorialIter(int n) {
        if (n < 0) return -1;
        long result = 1;
        for (int i = 2; i <= n; i++) {
            result *= i;
        }
        return result;
    }
    
    public static void main(String[] args) {
        int n = 5;
        
        System.out.println(n + "! = " + factorial(n));
        System.out.println(n + "! = " + factorialIter(n) + " (iterative)");
    }
}
